packages/flutter/lib/src/rendering/table.dart

```diff
@@ -385,76 +385,48 @@ class RenderTable extends RenderBox {
        assert(configuration != null),
        _textDirection = textDirection,
        _columns = columns ?? (children != null && children.isNotEmpty ? children.first.length : 0),
-       _rows = rows ?? 0,
+       _rows = rows ?? (children != null ? children.length : 0),
        _columnWidths = columnWidths ?? HashMap<int, TableColumnWidth>(),
        _defaultColumnWidth = defaultColumnWidth,
        _border = border,
        _textBaseline = textBaseline,
        _defaultVerticalAlignment = defaultVerticalAlignment,
        _configuration = configuration {
-    _children = <RenderBox?>[]..length = _columns * _rows;
     this.rowDecorations = rowDecorations; // must use setter to initialize box painters array
-    children?.forEach(addRow);
+    if (children != null) {
+      final List<RenderBox> allChildren = <RenderBox>[];
+      for (final List<RenderBox> rows in children)
+        rows.forEach(allChildren.add);
+
+      allChildren.reversed.forEach(insertChild);
```

Comment:
given that all that this does is insert(0) into _children, why not just add all the children to _children directly instead of adding to a local variable and then reversing the list (expensive) and then inserting all the children into the other list (expensive)?